File: fermionic-su-n/dima scripts/overlap.m
function S = overlap(z,m)

e(5:5)=0;
e(1,1)=z(1);
for i=2:5
e(i,1)=e(i-1,1)+z(i);
end
for n=2:5
for k=2:n
e(n,k)=e(n-1,k)+z(n)*e(n-1,k-1);
end
end
S=e(5,m);

File: fermionic-su-n/dima scripts/python/overlap.py
import numpy as np

def overlap(z, m):
    """
    z : array of length 5
    m : integer
    """
    e = np.zeros((5, 5), dtype=complex)

    e[0, 0] = z[0]
    for i in range(1, 5):
        e[i, 0] = e[i-1, 0] + z[i]

    for n in range(1, 5):
        for k in range(1, n+1):
            e[n, k] = e[n-1, k] + z[n] * e[n-1, k-1]

    return e[4, m-1]   # Matlab is 1-based

